### DISCUSSION TITLE
experimental warnings for asset checks

### DISCUSSION THREAD

**Discussion by geoHeil** [dagster-io/dagster#17520] [Q&A] [answered]
I am trying to ignore experimental warnings

```
import warnings

from dagster import ExperimentalWarning
import dagster

warnings.filterwarnings("ignore", category=ExperimentalWarning)
warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning)
```

however, dagster keeps logging:

```
ExperimentalWarning: Class `AssetCheckKey` is experimental. It may break in future versions, even between dot releases. To mute warnings for experimental functionality, invoke warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning) or use one of the other methods described at https://docs.python.org/3/library/warnings.html#describing-warning-filters.
```

how can I successfully ignore this?

**Comment by DennisSchwartz** [comment]
Same issue here. It works for other `ExperimentalWarning ` warnings, but this one seems to be exempt somehow.

**Comment by johannkm** [comment, marked as answer]
This is fixed by https://github.com/dagster-io/dagster/pull/17598, which will go in the release next week

**Comment by DennisSchwartz** [comment]
Hey hey! Thanks for addressing this. However, I just upgraded to `1.5.7` and I'm still seeing this issue.

I'm using 
```
warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning)
```
in the file where I create the asset definitions and it seems to turn off other ExerimentalWarning logs, but not the asset check ones.

Do I need to wait for 1.5.8 or am I doing something else wrong?

**Reply by johannkm** [reply]
Are you using 1.5.7 in all of the the webserver, daemon, and job code?

**Reply by johannkm** [reply]
And is the experimental log you're seeing for `AssetCheckKey` or something else?

**Reply by DennisSchwartz** [reply]
Hey Johann, thanks for getting back to me!

I'm getting these errors in the dozens and they seem to be related to `AssetCheckSpec`  - which I just now realise is slightly different from `AssetCheckKey`!

```
ExperimentalWarning: Class `AssetCheckSpec` is experimental. It may break in future versions, even between dot releases. To mute warnings for experimental functionality, invoke warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning) or use one of the other methods described at https://docs.python.org/3/library/warnings.html#describing-warning-filters.
  AssetCheckSpec(
  ```
  
I'm using Poetry for package management, so I've updated the Dagster version in my `pyproject.toml`, deleted the `poetry.lock` file and run `poetry install`.
  
  Here are the outputs of `poetry show` and `pip freeze`:
  
  
  ```
$ poetry show | grep dags                                                                                   
dagster                1.5.7        Dagster is an orchestration platform fo...
dagster-aws            0.21.7       Package for AWS-specific Dagster framew...
dagster-cloud          1.5.7        
dagster-cloud-cli      1.5.7        
dagster-graphql        1.5.7        The GraphQL frontend to python dagster.
dagster-pipes          1.5.7        Toolkit for Dagster integrations with t...
dagster-webserver      1.5.7        Web UI for dagster.

$ pip freeze | grep dagster
dagster==1.5.7
dagster-aws==0.21.7
dagster-cloud==1.5.7
dagster-cloud-cli==1.5.7
dagster-graphql==1.5.7
dagster-pipes==1.5.7
dagster-webserver==1.5.7
```

I guess I assumed `AssetCheckSpec` would also be affected by the update in this issue, but if not I apologise.

**Reply by johannkm** [reply]
I'll take a look if I can repro. Do you have a stack trace with the warning?

**Reply by DennisSchwartz** [reply]
Ok I'm sorry, it was my fault. I found the issue. I had another look at all the warnings I'm getting and realised I had warnings for `observable_source_assets` which I had previously suppressed successfully.

I think in my quest of silencing the `asset_check` errors I added the warnings filter to many different files and then removed them again from all places but one - it seems it wasn't the same as before.

In case this is useful to someone else in the future, this is the statement I'm using:

```
import warnings
import dagster
# Turn off Experimental Features warning for now
warnings.filterwarnings("ignore", category=dagster.ExperimentalWarning)
```

Adding it to the file which contains the instantiation of our asset definitions
```
# Initialise definitions
definitions = Definitions(
    assets=[*bronze_assets, *silver_assets, *gold_assets],
    resources={
        "file_manager": get_file_manager(),
        "entity_registry": entity_registry,
    },
)
```
means it's not being picked up!

Moving this statement to some other file where we keep the actual asset ops themselves has fixed.


Thanks again for your help, and sorry for the confusion! :)
